### PR TITLE
Ds/kafka wrapper

### PR DESCRIPTION
**Pull request by davesullivan41**
Added kafka wrapper for ibex for dramatic performance improvements, up to ~70 messages per second.

### CODE REVIEW

File: kafka_wrapper/app.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+def consume(queue, logger):
+    # Set up consumer
+    consumer = get_kafka_consumer(CONSUME_TOPIC, KAFKA_BROKERS)
+    logger.info("Set up to consume")
+    for message in consumer:
+        parsed_message = json.loads(message.value.decode("utf-8"))
+        queue.put(parsed_message)
+
+def dummy_consume(queue, logger):
```
Comment (poffdeluxe) [inline kafka_wrapper/app.py]: When/why does the dummy consume get used?

Also, after this function, there's some extra newlines, it looks like

File: kafka_wrapper/app.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        producer.send(PRODUCE_TOPIC, message.value)
+
+
+
+def get_topics(queue, logger):
+    # Set up producer and ibex
+    producer = get_kafka_producer(KAFKA_BROKERS)
+    ibex = EntityParser()
+    logger.info("Set up to extract topics")
+    while not queue == None:
```
Comment (bstarling) [inline kafka_wrapper/app.py]: Any way to rephrase this so it's not a double negative not == None is hard to parse mentally.

File: kafka_wrapper/app.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+ENABLE_IBEX = str(os.getenv("ENABLE", True))
+
+@retry(wait_exponential_multiplier=1000, wait_exponential_max=10 * 1000)
+def get_kafka_client(hosts):
+    return KafkaClient(hosts=hosts)
+
+@retry(wait_exponential_multiplier=1000, wait_exponential_max=10 * 1000)
+def get_kafka_consumer(topic, bootstrap_servers):
+    return KafkaConsumer(topic, group_id=ENABLE_IBEX, bootstrap_servers=bootstrap_servers)
```
Comment (bstarling) [inline kafka_wrapper/app.py]: Group ID should be a consumer group name. This is stored in kafka between consumer disconnecting and reconnecting so it can start reading from where it left off. We use like `quorum_<service>` so `quorum_ibex`

File: requirements.txt [hunk 4]
```diff
@@ -2,4 +2,6 @@ spacy>=2.0.11
 flask>=1.0.2
 markdown>=2.6.11
 pytest>=3.6.2
-pandas>=0.23.1
\ No newline at end of file
+pandas>=0.23.1
```
Comment (bstarling) [inline requirements.txt]: I know it's not related to your changes but is pandas still used? Could we remove it?

File: kafka_wrapper/app.py (hunk 2 above)
Comment (davesullivan41) [reply kafka_wrapper/app.py]: Yup, this was leftover from some early plans (that weren't going to work anyway). Fixed now.

File: kafka_wrapper/app.py (hunk 3 above)
Comment (davesullivan41) [reply kafka_wrapper/app.py]: Yeah, this was accidentally left in from when I was testing (I had one group that was disabled labeled 'false' and another that was enabled labelled 'true'). Fixed in upcomign commit

File: kafka_wrapper/app.py (hunk 1 above)
Comment (davesullivan41) [reply kafka_wrapper/app.py]: this is in case we'd like to stop ibex if it is causing problems. It's tied to an environment variable 'ENABLE'

File: requirements.txt (hunk 4 above)
Comment (davesullivan41) [reply requirements.txt]: Yup, removed
